apps/studio/components/interfaces/TableGridEditor/SidePanelEditor/TableEditor/RLSManagement/RLSManagement.tsx

```diff
@@ -0,0 +1,401 @@
+          <p className="text-sm text-foreground-lighter">No policies exist for this table</p>
+        </CardContent>
+      )
+    }
+
+    const handleRemoveGeneratedPolicy = (index: number) => {
+      // Find the index in generatedPoliciesList that corresponds to the allPoliciesList index
+      const generatedIndex = index - existingPoliciesList.length
+      if (generatedIndex >= 0 && generatedIndex < generatedPolicies.length) {
+        const updatedPolicies = generatedPolicies.filter((_, i) => i !== generatedIndex)
+        onGeneratedPoliciesChange?.(updatedPolicies)
+
+        // Track policy removal
+        if (project?.ref) {
+          track('rls_generated_policy_removed')
+        }
+      }
+    }
+
+    return (
+      <PolicyList
+        policies={allPoliciesList}
+        className="border-0 rounded-none"
+        onRemove={isNewRecord && !isDuplicating ? handleRemoveGeneratedPolicy : undefined}
+      />
+    )
+  }
+
+  if (!project) return null
+
+  if (isNewRecord && !isDuplicating) {
+    return (
+      <div>
+        <div className="flex items-center mb-4 gap-2">
+          <div className="flex-1">
+            <h3>Policies</h3>
+            <p className="text-sm text-foreground-lighter">
+              Set rules around who can read and write data to this table
+            </p>
+          </div>
+        </div>
+
+        <Card>{renderPolicies()}</Card>
+        {generatedPolicies?.length > 0 && (
+          <p className="text-xs text-foreground-lighter mt-4 flex items-center gap-2">
+            <InfoIcon size={16} strokeWidth={1.5} />
+            Review the policies generated before creating your table
+          </p>
+        )}
+        <AIOptInModal visible={isOptInModalOpen} onCancel={() => setIsOptInModalOpen(false)} />
+      </div>
+    )
+  }
+
+  return (
+    <div>
+      <div className="flex items-center mb-4 gap-2">
+        <div className="flex-1">
+          <h3>Policies</h3>
+          <p className="text-sm text-foreground-lighter">
+            Set rules around who can read and write data to this table
+          </p>
+        </div>
+        <Button
+          asChild
+          type="default"
+          size="tiny"
+          icon={<ExternalLink size={16} strokeWidth={1.5} />}
+        >
+          <Link href={`/project/${project.ref}/auth/policies`} target="_blank">
+            Manage policies
+          </Link>
+        </Button>
+      </div>
+
+      {isExistingTable && !rlsEnabled && (
+        <Admonition
+          className="mb-4"
+          type="warning"
+          title="Row Level Security is disabled"
+          description="Your table is currently accessible by anyone on the internet. We recommend enabling RLS to restrict access."
+          actions={renderEnableRlsButton()}
+        />
+      )}
+
+      <Card
+        aria-disabled={disablePoliciesList}
+        className={cn(disablePoliciesList && 'opacity-50 pointer-events-none')}
+      >
+        {renderPolicies()}
```

Comment:
this is quite confusing here - `renderPolicies` is doing too much IMO as it's also controlling the empty state, but within this return statement its no longer the empty state